sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/ClientRetryPolicy.java

```diff
@@ -116,8 +117,8 @@ public ClientRetryPolicy(GlobalEndpointManager globalEndpointManager,
 
     private ShouldRetryResult shouldRetryOnSessionNotAvailable() {
         this.sessionTokenRetryCount++;
-
         if (!this.enableEndpointDiscovery) {
+            logger.warn("SessionNotAvailable: no retry due to enableEndpointDiscovery=false");
```

Comment:
Since these are specifically for Walmart, why not use DEBUG and not warn, since warning logs will be difficult to diagnose, as they are a lot.

Reply:
DEBUG is not feasible.